pkg/core/middleware/authn.go

```diff
@@ -33,9 +34,12 @@ func (m *AuthnMiddleware) Handle(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		var err error
 		defer func() {
-			if err != nil {
-				// clear session if error occurred
+			if err == session.ErrSessionNotFound {
+				// clear session if session is not found
 				m.SessionWriter.ClearSession(w)
+			} else if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Some questions, 
1. I think we should provide the error message for debugging. If we don't want to expose the error message to client, maybe we should log the error? Otherwise, it is difficult to debug.
2. Can the SDK handle this kind of error? Without the response body.

Reply:
1. Yes, I've updated to log the error.
2. I've opened an issue to handle it in JS SDK.